Typo3 12 compatibility

hojalatheef opened the pull request.

hojalatheef committed:
[TASK] Compatibility Fixes and Removing support for older versions

- Conpatibility fix for TYPO3 version 12 has been done
- Removed older TYPO3 Version support

hojalatheef committed:
Documentation format changes and added Version details

hojalatheef committed:
Invalid lines removed from .gitattributes file

hojalatheef committed:
README.md modified with more informations

hojalatheef committed:
Language file formatting

hojalatheef committed:
Icons moved to Configuration/Icon.php and added .editorconfig and .crowdin.yml

hojalatheef committed:
Removec TSconfig inclusion from ext_localconf and added Configuration/page.tsconfig

hojalatheef committed:
Removec TSconfig inclusion from ext_localconf and added Configuration/page.tsconfig

hojalatheef committed:
TCA Configuration Fixes for TYPO3 Version 12 LTS

hojalatheef committed:
Moved functionality related to Provider initialization to Traits

hojalatheef committed:
SessionUtility added for managing frontend session related functionalities

hojalatheef committed:
[TASK] Refactoring functionality of Media Consent Provider

hojalatheef committed:
Interface addeed for the MediaConsent processor

hojalatheef committed:
[TASK] Code Formatting done for TypoScript files

hojalatheef committed:
[FEATURE] Documentation updates and jQuery replaced

- Documentation updated
- Removed jQuery with vanilla JavaScript

hojalatheef committed:
Removed null pointer warning

hojalatheef committed:
[TASK] GitIgnore added and crowdin file renamed as per new standards

hojalatheef committed:
[TASK] Added UnitTest for the Utility class and DataProcessor

hojalatheef committed:
[TASK] GitHub Action added for CI

hojalatheef committed:
Removed CI Functional Executions and added crowdin.yml to gitattributes

hojalatheef committed:
Label correction for Test Suite

sfroemkenjw commented on .github/workflows/testscorev12.yml:
```diff
@@ -0,0 +1,45 @@
+name: CI
```
"TYPO3 12", else in README.md the badge will just show "Test:CI"

hojalatheef replied:
As the new version supports only 12 LTS. I just added CI so the badge is fully rendered without any issues. I revert to TYPO3 12.

sfroemkenjw commented on .github/workflows/testscorev12.yml:
```diff
@@ -0,0 +1,45 @@
+      matrix:
+        php:
+          - '8.1'
+          - '8.2'
```
TYPO3 12 is officially PHP 8.3 compatible. What do you think? Should we start activating that, too?

hojalatheef replied:
Done

sfroemkenjw commented on .gitignore:
```diff
@@ -1,3 +1,37 @@
-.vscode
-# ignore generated documentation
+########################
+# events2
```
I'm pretty sure that's wrong ;-)

sfroemkenjw commented on Classes/DataProcessing/MediaconsentCnsProcessor.php:
```diff
@@ -9,80 +11,76 @@
  * Class for data processing for the content element Mediaconsent Cns
  */
-class MediaconsentCnsProcessor implements DataProcessorInterface
+class MediaconsentCnsProcessor implements MediaconsentCnsProcessorInterface
```
I don't understand this "Cns" thingy. As this is a major release, can we get rid of that?

sfroemkenjw commented on Classes/Traits/ProviderInitializationTrait.php:
```diff
@@ -0,0 +1,40 @@
+        try {
+            $cnProviders = ArrayUtility::getValueByPath($processorConfiguration, 'cnProviders') ?? false;
+            if ($cnProviders) {
+                $this->smcProviders = explode(',', $cnProviders);
```
If would prefer GeneralUtility::trimExplode(',', $cnProviders, true) as this is a human entered value

sfroemkenjw commented on Classes/Utility/SessionUtility.php:
```diff
@@ -0,0 +1,51 @@
+use Psr\Http\Message\ServerRequestInterface;
+use TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication;
+
+class SessionUtility
```
Please enter a short ClassHeader

sfroemkenjw commented on Classes/Utility/SessionUtility.php:
```diff
@@ -0,0 +1,51 @@
+    public static function setFrontendSessionData(ServerRequestInterface $request, string $key, array $data): void
+    {
+        $frontendUserAuthentication = self::getFrontendUserAuthentication($request);
+        if ($frontendUserAuthentication) {
```
Please check against instanceof FrontendUserAuthentication

sfroemkenjw commented on Configuration/TCA/Overrides/tt_content.php:
```diff
@@ -23,46 +24,46 @@
 ];
-\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTCAcolumns('tt_content',
+ExtensionManagementUtility::addTCAcolumns('tt_content',
     $tempColumns);
```
It's short enough. Just merge these both lines

sfroemkenjw commented on Configuration/TSconfig/PageTSConfig.tsconfig:
```diff
@@ -0,0 +1,20 @@
+mod.wizards.newContentElement.wizardItems.special {
```
Last time you had a cool filename like "NewContentElementWizard.tsconfig" for that file."

sfroemkenjw commented on Configuration/page.tsconfig:
```diff
@@ -0,0 +1,2 @@
+# Add mediaconsent_cns to new content element wizard
+@import 'EXT:mediaconsent/Configuration/TSconfig/PageTSConfig.tsconfig'
```
Update filename here, too

sfroemkenjw commented on Documentation/.DS_Store:
Something with your .gitignore is not working

sfroemkenjw replied:
And please remove that file from GIT

sfroemkenjw commented on Documentation/Index.rst:
```diff
@@ -1,49 +1,64 @@
-=============================================================
+======================
 Mediaconsent Extension
-=============================================================
```
Please remove the "Extension" wording. Of cause it's an "Extension"

sfroemkenjw commented on Documentation/Index.rst:
```diff
@@ -1,49 +1,64 @@
+    mediaconsent
 
+:Package name:
+    Media Consent
```
Composer package name: https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/GeneralConventions/FileStructure.html#package-name

sfroemkenjw commented on Resources/Public/js/tx_mediaconsent.js:
mediaconsent.js would be enough

sfroemkenjw commented on Resources/Public/js/tx_mediaconsent.js:
```diff
@@ -1,25 +1,40 @@
-var mcNsp = (function namespace() {
+let clickHandler = function (event) {
```
You're in JS global context. Because of possible duplicated variable usage I prefer to prepend `mc` to all variables in global context

sfroemkenjw commented on Resources/Public/js/tx_mediaconsent.js:
```diff
@@ -1,25 +1,40 @@
-    return {
-        reloadItem: reloadItem
+    this.reload = function () {
+      let xhr = new XMLHttpRequest();
```
Please use "fetch"

sfroemkenjw commented on Resources/Public/js/tx_mediaconsent.js:
```diff
@@ -1,25 +1,40 @@
+let clickHandler = function (event) {
+  let scpType = this.getAttribute('data-mctype');
+  let mediaconsentWrappers = document.querySelectorAll('.mediaconsent_wrapper[data-mctype="' + scpType + '"]');
+  mediaconsentWrappers.forEach(function (mediaconsentWrapper) {
```
Arrow function

sfroemkenjw commented on Resources/Public/js/tx_mediaconsent.js:
```diff
@@ -1,25 +1,40 @@
-            item.html(data);
-       }, 'html');
-    };
+var mcNsp = (function () {
```
Ue Arrow function

sfroemkenjw commented on Resources/Public/js/tx_mediaconsent.js:
```diff
@@ -1,25 +1,40 @@
-    });
-});
\ No newline at end of file
+window.onload = function () {
```
Use arrow function

sfroemkenjw commented on Resources/Public/js/tx_mediaconsent.js:
```diff
@@ -1,25 +1,40 @@
-});
\ No newline at end of file
+window.onload = function () {
+  var mediaconsentWrappers = document.getElementsByClassName('mediaconsent_wrapper');
```
USe `document.querySelector`

sfroemkenjw commented on Tests/Unit/DataProcessing/MediaconsentCnsProcessorTest.php:
```diff
@@ -0,0 +1,46 @@
+use JWeiland\Mediaconsent\DataProcessing\MediaconsentCnsProcessor;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+
+
```
Remove one empty row. Why is php-cs-fixer not complaining about that?

hojalatheef replied:
Nope

sfroemkenjw commented on Tests/Unit/Utility/SessionUtilityTest.php:
```diff
@@ -0,0 +1,100 @@
+use TYPO3\CMS\Core\Session\UserSession;
+use TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication;
+
+class SessionUtilityTest extends TestCase
```
Isn't there a UnitTestCase of TYPO3 TestingFramework?!

hojalatheef replied:
AI Assistant failed here and I didn't noticed that

sfroemkenjw commented on ext_localconf.php:
```diff
@@ -1,16 +1,7 @@
-\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPageTSConfig(
-    '<INCLUDE_TYPOSCRIPT: source="FILE:EXT:mediaconsent/Configuration/PageTS/PageTSConfig.tsconfig">'
-);
+if (!defined('TYPO3')) {
```
Remove file completely

hojalatheef committed:
[TASK] Refactoring and Optimizations

- Changed PHP Unit TestCase to TYPO3 UnitTestCase
- Renamed MediaconsentCnsProcessorTest to MediaConsentProcessorTest
- Renamed javascript asset file to mediaconsent.js
- Deleted empty ext_localconf.php
- Documentation corrections
- Deleted ignored DS file

hojalatheef committed:
[TASK] Refactoring and Optimizations

- Code Formatting Issues
- Javascript Modernizations
- File Ranamed to better

hojalatheef committed:
[BUGFIX] Modernization with small BUGFIX as the Response Object is called with text() method

hojalatheef committed:
[TASK] Update runTests.sh to support PHP 8.3 Testing
